sdks/java/core/src/main/java/org/apache/beam/sdk/testing/BigqueryMatcher.java

```diff
@@ -121,8 +121,8 @@ protected boolean matchesSafely(PipelineResult pipelineResult) {
       throw new RuntimeException("Failed to fetch BigQuery data.", e);
```

Comment:
I think it needs to re-interrupt the current Thread, similar as:
https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/util/GcsUtil.java#L191